functional/FunctionalTest.java

```diff
@@ -1074,11 +1074,17 @@ public static void copyObject_test7() throws Exception {
 
     copyConditions.setUnmodified(dateRepresentation);
 
+    ErrorResponseException expectedException = null;
+
     try {
       client.copyObject(bucketName, fileName, secondBucketName, copyConditions);
     } catch (ErrorResponseException e) {
       // File should not be copied as object was modified after date set in copyConditions.
-      ignore();
+      expectedException = e;
+    }
+
+    if (expectedException == null || !expectedException.errorResponse().code().equals("PreconditionFailed")) {
```

Comment:
copyObject_test7() is intended to test if copyObject() is always raising an exception when we pass an unsatisfying `x-amz-copy-source-if-unmodified-since` condition. We have a bug that this PR is actually fixing but this test wasn't actually detecting it before.